great_expectations/zep/postgres_datasource.py

```diff
@@ -34,7 +35,7 @@ class BatchRequestError(Exception):
 _DEFAULT_MONTH_RANGE = range(1, 13)
 
 
-@dataclasses.dataclass(frozen=True)
+@pydantic_dc.dataclass(frozen=True)
```

Comment:
This is a drop-in replacement for a `dataclass` but with type enforcement.
Also needed in order to specify these splitters as part of config yaml files.